flocx_market/db/sqlalchemy/api.py

```diff
@@ -38,96 +38,124 @@ def drop_db():
     return True
 
 
-def offer_get(marketplace_offer_id):
-    return get_session().query(models.Offer).filter_by(
-        marketplace_offer_id=marketplace_offer_id).first()
+def offer_get(marketplace_offer_id, context):
 
+    if context.is_admin:
+        return get_session().query(models.Offer).filter_by(
+            marketplace_offer_id=marketplace_offer_id).first()
+    else:
+        return get_session().query(models.Offer).filter_by(
+            marketplace_offer_id=marketplace_offer_id,
+            project_id=context.project_id).first()
```

Comment:
Thinking about it, I don't think Offer get/get_all needs to be scoped to project; it's legitimate for a consumer to browse all offers. I would remove this scoping from the get methods, and put it in the modify methods.

Reply:
Should there be a get all of my offers as well as a get all offers in the db api? Or is that kind of filtering something that should happen at the ui level

Reply:
Good question. I might suggest a 'get_all_by_project_id' function.

Reply:
Should this be part of the flask api?

Reply:
There would need to be 2 parameters in the get function. This also goes against the api documentation sahil has provided, doesn't it?

Reply:
For now, I'd suggest creating the get_all_by_project_id function in the database API and the business object layer, but leaving the REST API alone. We can decide later what we want the exact behavior to be.

I'd consider the API documentation something of a "living document" : )